addons/purchase/tests/test_purchase_order.py

```diff
@@ -83,3 +84,82 @@ def test_00_purchase_order_flow(self):
+
+    def test_01_vendor_price(self):
+        #Case1: We find a vendor for the product variant and take the variant vendor price.
+        #Add the vendors in product variants.
+        supplier_info_1 = {
+            'product_seller_ids': [
+                (0, 0, {
+                    'name': self.partner_id.id,
+                    'min_qty': 5.0,
+                    'price': 500.0,
+                })
+            ]
+        }
+        self.product_id_1.write(supplier_info_1)
+
+        #Lets create a purchase order to test the vendor price.
+        po_vals_1 = {
+            'partner_id': self.partner_id.id,
+        }
+        self.po1 = self.PurchaseOrder.create(po_vals_1)
+
+        po_line_dict_1 = {
+            'name': self.product_id_1.name,
+            'product_id': self.product_id_1.id,
+            'order_id': self.po1.id,
+            'product_uom': self.product_id_1.uom_po_id.id,
+            'date_planned': datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+        }
+
+        #Call the 'onchange_product_id' to get the price from vendor.
+        po_line_1 = self.PurchaseOrderLine.new(po_line_dict_1)
+        po_line_1.onchange_product_id()
+        line_dict_1 = po_line_1._convert_to_write({key: po_line_1[key] for key in po_line_1._cache})
+        self.order_line1 = self.PurchaseOrderLine.create(line_dict_1)
+
+        self.assertEqual(self.order_line1.product_qty, 5.0, 'Purchase: the minimal quantity of the product for the supplier should be 5.0.')
+        self.assertEqual(self.order_line1.price_unit, 500.0, 'Purchase: the price of the product for the supplier should be 500.0.')
+
+        #Case2: When we don't find a vendor for the product variant, we take the price from product template.
+        #Add the vendors in product variants and product template.
+        supplier_info_2 = {
+            'product_seller_ids': [
+                (0, 0, {
+                    'name': self.partner_id.id,
+                    'min_qty': 10,
+                    'price': 100,
+                })
+            ],
+            'seller_ids': [
+                (0, 0, {
+                    'name': self.partner_id_1.id,
+                    'min_qty': 4,
+                    'price': 150,
+                })]
+        }
+        self.product_id_1.write(supplier_info_2)
+
+        #Lets create a purchase order to test the vendor price.
+        po_vals_2 = {
+            'partner_id': self.partner_id_1.id,
+        }
+        self.po2 = self.PurchaseOrder.create(po_vals_2)
+
+        po_line_dict_2 = {
+            'name': self.product_id_1.name,
+            'product_id': self.product_id_1.id,
+            'order_id': self.po2.id,
+            'product_uom': self.product_id_1.uom_po_id.id,
+            'date_planned': datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+        }
+
+        #Call the 'onchange_product_id' to get the price from vendor.
+        po_line_2 = self.PurchaseOrderLine.new(po_line_dict_2)
```

Comment:
I used **new()**, to create a draft record to perform on-change. Here i need **Unit Price** from that onchange method.